### PR TITLE
Support Shanghai in Reference Tests

### PR DESCRIPTION
**Pull request by shemnon**
## PR description

Add support for Shanghai in reference tests
* Add Shanghai and Cancun milestones
* Add "withdrawals" and "withdrawalsRoot" to json objects
* Migrate to schedule by block header.

Signed-off-by: Danno Ferrin <danno.ferrin@swirldslabs.com>

## Documentation

- [ ] I thought about documentation and added the `doc-change-required` label to this PR if
    [updates are required](https://wiki.hyperledger.org/display/BESU/Documentation).

## Changelog

- [ ] I thought about the changelog and included a [changelog update if required](https://wiki.hyperledger.org/display/BESU/Changelog).

### CODE REVIEW

File: ethereum/referencetests/src/main/java/org/hyperledger/besu/ethereum/referencetests/BlockchainReferenceTestCaseSpec.java
Comment (macfarla) [inline ethereum/referencetests/src/main/java/org/hyperledger/besu/ethereum/referencetests/BlockchainReferenceTestCaseSpec.java:179]: is this a TODO?

**Review by macfarla — APPROVED** [review]
LGTM - I'm assuming the commented out lines will come in for withdrawals

File: ethereum/referencetests/src/main/java/org/hyperledger/besu/ethereum/referencetests/BlockchainReferenceTestCaseSpec.java
Comment (shemnon) [reply ethereum/referencetests/src/main/java/org/hyperledger/besu/ethereum/referencetests/BlockchainReferenceTestCaseSpec.java:179]: This is what withdrawals will do with it's code, once it's merged.

**Review by jframe — APPROVED** [review]
LGTM

**Review by siladu — APPROVED** [review]
Nice! I've made a note to revisit these comments on the Withdrawals Epic: https://github.com/hyperledger/besu/issues/4746

**Pull request merged**
